### ISSUE TITLE
tailscaled: move expiryManager node mutations up a layer or two

### ISSUE THREAD

**Issue by bradfitz** [tailscale/tailscale#8963]
While working on #8961 (itself a few yaks deep from my real goal), I noticed that the only thing which tried to mutate NetworkMap.Peers nodes (*tailcfg.Node) in the whole process is the `ipnlocal` `expiryManager`.

That code became this, somewhat unfortunately:

```
		mut := peer.AsStruct()

		// Actually mark the node as expired
		mut.Expired = true

		// Control clears the Endpoints and DERP fields of expired
		// nodes; do so here as well. The Expired bool is the correct
		// thing to set, but this replicates the previous behaviour.
		//
		// NOTE: this is insufficient to actually break connectivity,
		// since we discover endpoints via DERP, and due to DERP return
		// path optimization.
		mut.Endpoints = nil
		mut.DERP = ""

		// Defense-in-depth: break the node's public key as well, in
		// case something tries to communicate.
		mut.Key = key.NodePublicWithBadOldPrefix(peer.Key())

		netmap.Peers[i] = mut.View()
```

So I'm cloning a copy, mutating it, and mutating the original slice.

That means I can't pass in a immutable slice view in follow-up change, as there's this one caller that needs to write to it.

Perhaps we should move the `expiryManager` up a layer into the controlclient Direct client map streaming code that's reading from the network and building the NetMap in the first place.

Then everything downstream would be read-only?

/cc @andrew-d @maisem

**Comment by bradfitz** [comment]
Of course the whole point of expiryManager is to keep working even if control is down so controlclient isn't quite the right place for it either. Hmm.

**Comment by andrew-d** [comment]
> Of course the whole point of expiryManager is to keep working even if control is down so controlclient isn't quite the right place for it either. Hmm.

Yeah, that's why I didn't put it in `controlclient`. I wonder if we could have something between `LocalBackend.setClientStatus` and `controlclient.Options.Status`? i.e. instead of `Status: b.setClientStatus`, do `Status: b.em.makeClientStatusCb(b.setClientStatus)` or something...
